components/brave_wallet/browser/solana_instruction_builder.cc

```diff
@@ -3,54 +3,37 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this file,
  * You can obtain one at https://mozilla.org/MPL/2.0/. */
 
-#ifdef UNSAFE_BUFFERS_BUILD
-// TODO(https://github.com/brave/brave-browser/issues/41661): Remove this and
-// convert code to safer constructs.
-#pragma allow_unsafe_buffers
-#endif
-
 #include "brave/components/brave_wallet/browser/solana_instruction_builder.h"
 
 #include <optional>
-#include <type_traits>
 #include <utility>
 
+#include "base/containers/extend.h"
+#include "base/containers/span.h"
+#include "base/numerics/byte_conversions.h"
 #include "brave/components/brave_wallet/browser/simple_hash_client.h"
 #include "brave/components/brave_wallet/browser/solana_account_meta.h"
 #include "brave/components/brave_wallet/browser/solana_instruction.h"
 #include "brave/components/brave_wallet/common/brave_wallet.mojom.h"
 #include "brave/components/brave_wallet/common/brave_wallet_constants.h"
 #include "brave/components/brave_wallet/common/brave_wallet_types.h"
 #include "brave/components/brave_wallet/common/encoding_utils.h"
-#include "build/build_config.h"
+
+namespace brave_wallet {
+
 namespace {
 
-// Solana uses bincode::serialize when encoding instruction data, which encodes
-// unsigned numbers in little endian byte order.
-template <typename T>
-void UintToLEBytes(T val, std::vector<uint8_t>* bytes) {
-  static_assert(
-      std::is_same<T, uint64_t>::value || std::is_same<T, uint32_t>::value,
-      "Incorrect type passed to function UintToLEBytes.");
-
-  DCHECK(bytes);
-  size_t vec_size = sizeof(T) / sizeof(uint8_t);
-  *bytes = std::vector<uint8_t>(vec_size);
-
-  uint8_t* ptr = reinterpret_cast<uint8_t*>(&val);
-  for (size_t i = 0; i < vec_size; i++) {
-#if defined(ARCH_CPU_LITTLE_ENDIAN)
-    bytes->at(i) = *ptr++;
-#else
-    bytes->at(vec_size - 1 - i) = *ptr++;
-#endif
-  }
-}
+constexpr auto kSetComputeUnitLimit = static_cast<uint8_t>(
+    mojom::SolanaComputeBudgetInstruction::kSetComputeUnitLimit);
+constexpr auto kSetComputeUnitPrice = static_cast<uint8_t>(
+    mojom::SolanaComputeBudgetInstruction::kSetComputeUnitPrice);
+constexpr auto kTransfer =
+    static_cast<uint8_t>(mojom::SolanaSystemInstruction::kTransfer);
+constexpr auto kTransferChecked =
+    static_cast<uint8_t>(mojom::SolanaTokenInstruction::kTransferChecked);
```

Comment:
Do we really need this? I feel this is just creating a bit of noise.

Reply:
I'm also a bit confused why we are setting up those values up there as `uint8_t` and then we assigned them down here as entirely different types:

```cxx
  base::Extend(instruction_data, base::U32ToLittleEndian(kTransfer));
  base::Extend(instruction_data, base::U64ToLittleEndian(lamport));
```

Reply:
I've moved these constants to a separate section to avoid verbosity where instructions_datas are built. 
Fixed `kTransfer` to be uint32.